test-framework/sudo-compliance-tests/src/sudoers/user_list.rs

```diff
@@ -76,3 +76,96 @@ fn group_id() -> Result<()> {
         .exec(&env)?
         .assert_success()
 }
+
+#[test]
+fn many_different() -> Result<()> {
+    let env = Env(format!("root, {USERNAME} ALL=(ALL:ALL) NOPASSWD: ALL"))
+        .user(USERNAME)
+        .build()?;
+
+    Command::new("sudo")
+        .arg("true")
+        .exec(&env)?
+        .assert_success()?;
+
+    Command::new("sudo")
+        .arg("true")
+        .as_user(USERNAME)
+        .exec(&env)?
+        .assert_success()
+}
+
+#[test]
+fn many_repeated() -> Result<()> {
+    let env = Env("root, root ALL=(ALL:ALL) NOPASSWD: ALL").build()?;
+
+    Command::new("sudo")
+        .arg("true")
+        .exec(&env)?
+        .assert_success()
+}
+
+#[test]
+fn double_negation_cancels() -> Result<()> {
+    let env = Env("!!root ALL=(ALL:ALL) NOPASSWD: ALL")
+        .user(USERNAME)
+        .build()?;
+
+    Command::new("sudo")
+        .arg("true")
+        .exec(&env)?
+        .assert_success()
+}
+
+#[test]
+fn negation_excludes_group_members() -> Result<()> {
+    let env = Env("%users, !ghost ALL=(ALL:ALL) ALL")
+        // use PAM to avoid `ghost` getting a password prompt
+        .file("/etc/pam.d/sudo", PAMD_SUDO_PAM_PERMIT)
+        // the primary group of all new users is `users`
+        .user("ferris")
+        .user("ghost")
+        .build()?;
+
+    Command::new("sudo")
+        .arg("true")
+        .as_user("ferris")
+        .exec(&env)?
+        .assert_success()?;
+
+    let output = Command::new("sudo")
+        .arg("true")
+        .as_user("ghost")
+        .exec(&env)?;
+
+    assert!(!output.status().success());
+    assert_eq!(Some(1), output.status().code());
+
+    if sudo_test::is_original_sudo() {
+        assert_contains!(output.stderr(), "ghost is not in the sudoers file");
+    }
+
+    Ok(())
+}
+
+#[test]
+fn negation_is_order_sensitive() -> Result<()> {
+    // the group item overrides the preceding negated item
```

Comment:
This comment is inaccurate, since the negated item is not negating anything that wasn't negated.
```suggestion
// negated items at the start of a specifier list have are meaningless
```